File: v3-sandbox/prototype-keys-java/src/main/java/org/hiero/keys/PublicKey.java
package org.hiero.keys;

import org.hiero.keys.impl.KeyFactory;
import org.hiero.keys.io.ByteImportEncoding;
import org.hiero.keys.io.KeyFormat;

import static org.hiero.keys.io.KeyFormat.SPKI_WITH_PEM;

public interface PublicKey extends Key {

    boolean verify(final byte[] message, final byte[] signature);

    static PublicKey create(final KeyAlgorithm algorithm, final byte[] rawBytes) {
        return KeyFactory.createPublicKey(algorithm, rawBytes);
    }

    static PublicKey create(final KeyAlgorithm algorithm, final ByteImportEncoding encoding, final String value) {
        return KeyFactory.createPublicKey(algorithm, encoding, value);
    }

    static PublicKey create(final KeyFormat container, final byte[] value) {
        return KeyFactory.createPublicKey(container, value);
    }

    static PublicKey create(final KeyFormat container, final String value) {
        return KeyFactory.createPublicKey(container, value);
    }

    static PublicKey create(final String value) {
        return create(SPKI_WITH_PEM, value);
    }

}

File: v3-sandbox/prototype-python/src/hiero_sdk_v3/keys/public_key.py
from __future__ import annotations
from abc import abstractmethod
from .key import Key
from .key_algorithm import KeyAlgorithm
from .io.key_format import KeyFormat


class PublicKey(Key):
    @abstractmethod
    def verify(self, message: bytes, signature: bytes) -> bool:
        ...

    @staticmethod
    def create(algorithm: KeyAlgorithm, raw_bytes: bytes) -> PublicKey:
        from ._impl import create_public_key_from_raw
        return create_public_key_from_raw(algorithm, raw_bytes)

    @staticmethod
    def create_from_format(format: KeyFormat, value: bytes | str) -> PublicKey:
        from ._impl import create_public_key_from_format
        return create_public_key_from_format(format, value)

    @staticmethod
    def from_pem(pem: str) -> PublicKey:
        return PublicKey.create_from_format(KeyFormat.SPKI_WITH_PEM, pem)
